sympy/polys/matrices/sdm.py

```diff
@@ -853,8 +853,12 @@ def is_lower(self):
         """
         return all(i >= j for i, row in self.items() for j in row)
 
-    def lll(A, delta=QQ(3, 4)) -> 'SDM':
-        return A.from_ddm(ddm_lll(A.to_ddm(), delta))
+    def lll(A, delta=QQ(3, 4)):
+        return A.from_ddm(_ddm_lll(A.to_ddm(), delta=delta)[0])
```

Comment:
It would be better to make separate functions `ddm_lll` and `ddm_lll_transform` so that they can be used here without knowledge of the `lll` module's internal `_ddm_lll` function. This method computes the transformation matrix but discards it which is a little slower than not computing the transformation matrix.

Reply:
Oh, I see it's actually discarding the None. In any case it would still be clearer for the `lll` module to export functions `ddm_lll` and `ddm_lll_transform`. The fact that those both call `_ddm_lll` and in one case the transformation matrix is None should be hidden as an implementation detail of the `lll` module.